File: src/exercises/robot_simulator/typescript/template.ts
export type Direction = "north" | "east" | "south" | "west";
export type Coordinates = [number, number];

export class Robot {
  public get bearing(): Direction {
    // Your code here
    return "north";
  }

  public get coordinates(): Coordinates {
    // Your code here
    return [0, 0];
  }

  public place(position: { x: number; y: number; direction: Direction }): void {
    // Your code here
  }

  public evaluate(instructions: string): void {
    // Your code here
  }
}

File: src/exercises/robot_simulator/python/template.py
NORTH = "north"
EAST = "east"
SOUTH = "south"
WEST = "west"

class Robot:
    def __init__(self, direction=NORTH, x=0, y=0):
        self.direction = direction
        self.coordinates = (x, y)

    def turn_right(self):
        # Your code here
        pass

    def turn_left(self):
        # Your code here
        pass

    def advance(self):
        # Your code here
        pass

    def move(self, instructions):
        # Your code here
        pass
